utils/_context/containers.py

```diff
@@ -526,7 +526,11 @@ def load(self):
                 pytest.exit(f"Image {self.name} not found locally, please build it", 1)
 
             logger.stdout(f"Pulling {self.name}")
-            self._image = _get_client().images.pull(self.name)
+            try:
+                self._image = _get_client().images.pull(self.name)
+            except docker.errors.ImageNotFound:
+                # Sometimes pull returns ImageNotFound, internal race?
```

Comment:
I suggest adding a small sleep here, maybe 5s ? 

On the official system-tests workflow, we use a 3 times - 10s sleep time retry (plus a unique call to docker pull, which may be the real fix to that issue). With that, it has been month since I've seen this error : https://github.com/DataDog/system-tests/blob/main/.github/actions/pull_images/action.yml#L67-L68